askap/imageaccess/CasaADIOSImageAccessParallel.h

```diff
@@ -0,0 +1,274 @@
+/// @file CasaADIOSImageAccessParallel.h
+/// @brief Access casa image via ADIOS Storage Manager using parallel io
+/// @details This class implements IImageAccess interface for CASA image and specifies use of the ADIOS storage manager
+///
+///
+/// @copyright (c) 2007 CSIRO
+/// Australia Telescope National Facility (ATNF)
+/// Commonwealth Scientific and Industrial Research Organisation (CSIRO)
+/// PO Box 76, Epping NSW 1710, Australia
+/// atnf-enquiries@csiro.au
+///
+/// This file is part of the ASKAP software distribution.
+///
+/// The ASKAP software distribution is free software: you can redistribute it
+/// and/or modify it under the terms of the GNU General Public License as
+/// published by the Free Software Foundation; either version 2 of the License,
+/// or (at your option) any later version.
+///
+/// This program is distributed in the hope that it will be useful,
+/// but WITHOUT ANY WARRANTY; without even the implied warranty of
+/// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+/// GNU General Public License for more details.
+///
+/// You should have received a copy of the GNU General Public License
+/// along with this program; if not, write to the Free Software
+/// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
+///
+/// @author Alexander Williamson <alex.williamson@icrar.org>
+/// @author Pascal Jahan Elahi <pascal.elahi@pawsey.org.au>
+
+#ifndef ASKAP_ACCESSORS_CASA_ADIOS_IMAGE_ACCESS_PARALLEL_H
+#define ASKAP_ACCESSORS_CASA_ADIOS_IMAGE_ACCESS_PARALLEL_H
+
+#include <boost/shared_ptr.hpp>
+
+#include <askap/imageaccess/IImageAccess.h>
+#include <askap/imageaccess/ADIOSImage.h>
+#include <askap/imageaccess/CasaADIOSImageAccess.h>
+#include <mpi.h>
+#include <askap/askapparallel/MPIComms.h>
+#include <askap/askapparallel/AskapParallel.h>
+
+//PJE - still a work in progress 
+
+namespace askap {
+namespace accessors {
+
+/// @brief Access casa image via ADIOS2
+/// @details This class implements IImageAccess interface for CASA image via ADIOS2
+/// @ingroup imageaccess
+
+template <class T = casacore::Float>
+struct CasaADIOSImageAccessParallel : public CasaADIOS2ImageAccess<T> {
+
+public:
+    /// @brief constructor
+    /// @param[in] comms, MPI communicator
+    /// @param[in] config, string pointing to adios configuration
+    template <class T> CasaADIOSImageAccessParallel<T>(askapparallel::AskapParallel &comms, std::string configname = "");
+
+    // PJE -- uncertain how many functions would really require overwriting so as to 
```

Comment:
So far, the only ADIOSImage function that uses a comm is the makeNewTable function. This is only called via `CasaADIOSImageAccess<T>::create(...)' function.

This does raise the question about reading. The adios storage manager is used to read as well, but we don't need to initialise it for this as it's invoked automatically. Does this mean that it's using MPI for writing but not for reading? If so, do we need to somehow explicitly invoke it with a comm so we get parallel read as well?